lsp-protocol.el

```diff
@@ -246,6 +254,25 @@ Allowed params: %s" interface (reverse (-map #'cl-first params)))
          (apply #'append)
          (cl-list* 'progn))))
 
+(pcase-defmacro lsp-interface (interface &rest property-bindings)
```

Comment:
For `lsp-interface` (the one in `defmacro`), can you add the declaration form for it? Something like `(declare (indent 1))`.
That would make the indent for `lsp-interface`, both outside and inside `pcase` looks much better

Reply:
I do agree that would look nicer. Do you know if there a bunch of existing `lsp-interface`s that would need to be re-indented, besides ones we'd already be touching for the `pcase` change? (Just trying to avoid blowing up the scope of this PR too much.)

Reply:
Yuck. The problem with adding `(declare (indent 1))` or similar is that `lsp-interface` the macro actually takes multiple interface definitions at once. So, e.g., it winds up indented like

```lisp
(lsp-interface
    (eslint:StatusParams  (:state) nil)
  (eslint:OpenESLintDocParams (:url) nil)
  (eslint:ConfirmExecutionParams (:scope :file :libraryPath) nil))
```

Unless there's a strong opinion, I'd like to leave this alone for now&mdash;it's ugly, but I think the macro is used more frequently than the pcase form.

Reply:
Also, the `lsp-make-INTERFACE` functions kind of have the same problem, so maybe there could be a separate PR focused on indentation.